PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/toolkit/intellij/common/AzureArtifactComboBox.java

```diff
@@ -50,15 +49,10 @@ public void setFileFilter(final Condition<? super VirtualFile> filter) {
         this.fileFilter = filter;
     }
 
-    public synchronized void refreshItems(AzureArtifact defaultArtifact) {
-        unsubscribeSubscription(subscription);
-        this.setLoading(true);
-        subscription = this.loadItemsAsync()
-            .subscribe(items -> DefaultLoader.getIdeHelper().invokeLater(() -> {
-                this.setItems(items);
-                this.setLoading(false);
-                this.resetDefaultValue(defaultArtifact);
-            }), this::handleLoadingError);
+    public void setArtifact(@Nullable final AzureArtifact azureArtifact) {
+        final AzureArtifactManager artifactManager = AzureArtifactManager.getInstance(this.project);
+        this.cachedArtifact = azureArtifact;
+        this.setValue(new AzureComboBox.ItemReference<>(artifact -> artifactManager.equalsAzureArtifact(cachedArtifact, artifact)));
```

Comment:
this method may execute after items have been loaded, which causes the file artifact still not be added into item list.